### PR TITLE
Added ErrorBoundary HOC and component.

### PR DESCRIPTION
**Pull request by gabiwassan**
Hi team,
This PR includes the implementation of the ErrorBoundary component and HOC to catch components with errors on rendering.
Plus added `loglevel` dependency to log errors, warnings, etc.
e.g of error prompt:
image

Closes: #3313

### CODE REVIEW

File: public/components/common/hocs/error-boundary/with-error-boundary.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    render() {
+      debugger;
+      // @ts-ignore
+      if (this.state.errorInfo) {
+        return handleError(this);
+      }
+      return <WrappedComponent {...this.props} />;
+    }
+  };
```
Comment (Desvelao) [inline public/components/common/hocs/error-boundary/with-error-boundary.tsx:41]: I think that the HOC could be refactored to:
```ts
export const withErrorBoundary = WrappedComponent => props => (
  <ErrorBoundary>
    <WrappedComponent {...props} />
  </ErrorBoundary>
)
```

If the refactoring is possible, I would move the `catchFunc`  and `handleError` functions as components methods.

**Comment by Desvelao** [comment]
I suggest exporting the HOC in the `public/components/common/hocs`, so the HOC can be imported more easily.

File: public/components/common/hocs/error-boundary/with-error-boundary.tsx (hunk 1 above)
Comment (gabiwassan) [reply public/components/common/hocs/error-boundary/with-error-boundary.tsx:41]: Great!
Maybe we can use [this](https://www.npmjs.com/package/react-error-boundary)

File: public/components/common/hocs/error-boundary/with-error-boundary.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const catchFunc = (error, errorInfo, ctx) => {
+  ctx.setState({
+    error: error,
+    errorInfo: errorInfo,
+  });
+
+  log.error({ error, errorInfo });
+};
+
+const ErrorComponent = (props: { ctx: any }) => {
```
Comment (frankeros) [inline public/components/common/hocs/error-boundary/with-error-boundary.tsx:49]: it would be nice if was a child component for future enhancements

**Comment by frankeros** [comment]
please add unit test for the new components

File: public/components/common/hocs/error-boundary/with-error-boundary.tsx (hunk 2 above)
Comment (gabiwassan) [reply public/components/common/hocs/error-boundary/with-error-boundary.tsx]: Great!

File: public/components/common/hocs/error-boundary/with-error-boundary.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Wazuh app - React HOCs to manage user authorization requirements
```
Comment (Desvelao) [inline public/components/common/hocs/error-boundary/with-error-boundary.tsx:2]: Fix the file description

**Comment by Desvelao** [comment]
As a suggestion, you could add the license block to the new files (not to snapshots).

**Pull request merged**